src/main/scala/mesosphere/marathon/core/launchqueue/impl/LaunchQueueActor.scala

```diff
@@ -125,6 +120,17 @@ private[impl] class LaunchQueueActor(
       sender() ! None
   }
 
+  private[this] def receiveInstanceUpdateToSuspendedActor: Receive = {
+    case update: InstanceChange if suspendedLauncherPathIds(update.runSpecId) =>
+      // Do not defer. If an AppTaskLauncherActor restarts, it retrieves a new task list.
+      // If we defer this, there is a potential deadlock (resolved by timeout):
+      //   * AppTaskLauncher waits for in-flight tasks
+      //   * TaskOp gets processed and one of the update steps calls this here
+      //   * ... blocked until timeout ...
+      //   * The task launch notification (that the AppTaskLauncherActor waits for) gets sent to the actor
+      sender() ! None
```

Comment:
IIRC previously the message was stashed and eventually the ask timed out … basically what the comment is saying. What bothers me a little is that the QueuedInstanceInfo, which is sent back in the usual case, is not used by anyone, only in test classes (which could as well do a `Get`in order to retrieve the count). I'd be happy to change affected code to not ask or simply expect a Done or whatever – we're mostly mapping the response to Done or Unit.